### ISSUE TITLE
Ask only once for a given depency on apply-all & co

### ISSUE THREAD

**Issue by rgarrigue** [gruntwork-io/terragrunt#946]
Hi

So, I've something a bit like this

```
.
├── assumable-roles
│   ├── cicd
│   ├── development
│   └── production
├── groups
│   ├── admins
│   ├── cicd
│   └── developers
├── metadata
```

Each subfolder in `assumable-roles` have a dependency on `metadata` folder. If I run a `destroy-all` in `assumable-roles`, it'll ask [number of dependants subfolder] times the same question

> [terragrunt] 2019/11/07 15:37:33 [terragrunt]  Module /.../assumable-roles/development depends on module /.../metadata, which is an external dependency outside of the current working directory. Should Terragrunt run this external dependency? Warning, if you say 'yes', Terragrunt will make changes in /.../metadata as well! (y/n) 

I'ld like Terragrunt to ask only once for each unique dependency in the whole set of change. If possible.

**Comment by yorinasub17** [comment]
Thanks for reporting, but this is a duplicate of https://github.com/gruntwork-io/terragrunt/issues/521
